app/modules/topics/saga/task/fork.js

```diff
@@ -0,0 +1,23 @@
+// @flow
+
+import { type Saga } from 'redux-saga';
+import { call, put } from 'redux-saga/effects';
+
+import asyncRequests from 'modules/asyncRequests';
+import users from 'modules/users';
+
+import actions from '../../actions';
+import * as a from '../../actionTypes';
+
+const fork = function* (action: a.ForkAction): Saga<void> {
+  // Fork the topic in the backend
+  const { userId, topicId } = yield yield call(
+    asyncRequests.lib.putAndReturn,
+    actions.apiPostFork(action.payload.id),
+  );
+
+  // Add the topic to the user
+  yield put(users.actions.addTopicId(userId, topicId));
```

Comment:
- Adding the new topic to the current user instead of passing the userId in from the action is an unexpected behavior imo; if you want to keep it that way the action should be called `forkToCurrentUser` or similar.
- My personal preference would be to call this saga from the users module (similar to how `addTopic` and `removeTopic` are handled) and add the new topic to the user there. I just realized this doesn't cause an explicit error anymore since I refactored the rootSaga a few months ago (I will try to fix that, and I just realized that there are a few similar problems that I missed for this reason), but it's still the case that the topics module should not have a dependency on the users module and this is likely to cause problems down the line.

Reply:
It makes sense to move the concern to the users module.